activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -54,16 +54,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
```

Comment:
> Mutating this value (in the old code) is almost sure to result in incorrect behaviour, since the value is not retained by the receiver. How can you possibly defend this as desired behaviour?

I see. Is there any way of testing this in our test suite?